### PR TITLE
feat: Bundle using Rollup for multi-module support.

### LINKED ISSUE #139: Attribution imports not working with basic create-react-app usage
**Versions**
```
| => npm -v
10.5.0

| => node -v
v18.20.1

| => npm ls @honeycombio/opentelemetry-web
repro-app@0.1.0 /.../.../.../repro-app
└── @honeycombio/opentelemetry-web@0.2.0

| => npm ls @opentelemetry/auto-instrumentations-web
repro-app@0.1.0 /.../.../.../repro-app
└── @opentelemetry/auto-instrumentations-web@0.39.0
```

**Steps to reproduce**

1. `npx create-react-app repro-app --template typescript`
2. `cd repro-app`
3. `npm install @honeycombio/opentelemetry-web @opentelemetry/auto-instrumentations-web`
4. Add code to `src/index.tsx`:
```
import { HoneycombWebSDK, WebVitalsInstrumentation } from '@honeycombio/opentelemetry-web';
import { getWebAutoInstrumentations } from '@opentelemetry/auto-instrumentations-web';

const sdk = new HoneycombWebSDK({
  apiKey: 'api-key-goes-here',
  serviceName: 'your-great-browser-application',
  instrumentations: [getWebAutoInstrumentations(), new WebVitalsInstrumentation()], // add automatic instrumentation
});
sdk.start();
```
5. `npm start`

Browser console error:
```
web-vitals-autoinstrumentation.js:134 Uncaught TypeError: (0 , attribution_1.onCLS) is not a function
    at WebVitalsInstrumentation.enable (web-vitals-autoinstrumentation.js:134:1)
    at enableInstrumentations (autoLoaderUtils.ts:75:1)
    at registerInstrumentations (autoLoader.ts:42:1)
    at HoneycombWebSDK.start (base-otel-sdk.js:75:1)
    at ./src/index.tsx (index.tsx:14:1)
    at options.factory (react refresh:6:1)
    at __webpack_require__ (bootstrap:22:1)
    at startup:7:1
    at startup:7:1
```

**Additional context**

```
tsconfig.json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": [
    "src"
  ]
}

```

### PR DESCRIPTION
**Pull request by wolfgangcodes**
## Which problem is this PR solving?
Adds support for building  `esm` and `cjs` modules using rollup. Add an examples created using `react-create-app` and an example using cjs.

- Closes #139 

## How to verify that this has the expected result
- Build the sample apps
- Create a new app with `create-react-app` and install the web distro

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,20 +2,29 @@
...
+    "compile": "rollup -c",
+    "build:clean": "rm -rf dist",
+    "build": "npm run build:clean && npm run compile",
     "check-format": "prettier --list-different \"./**/*.{ts,mts,cts,js,cjs,mjs,tsx,jsx}\"",
     "format": "prettier --write \"./**/*.{ts,mts,cts,js,cjs,mjs,tsx,jsx}\"",
-    "clean": "tsc --build --clean",
     "lint": "eslint .",
     "lint:ci": "npm run lint:ci-prep && npm run lint",
-    "lint:ci-prep": "cd ./examples/hello-world-web && npm ci && cd ../custom-with-collector-ts/ && npm ci",
+    "lint:ci-prep": "cd ./examples/hello-world-web && npm ci && cd ../hello-world-cjs/ && npm ci && cd ../hello-world-cjs/ && npm ci && cd ../custom-with-collector-ts/ && npm ci",
```
Comment (wolfgangcodes) [inline package.json:27]: WG to fix.

File: rollup.config.mjs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }),
+  analyze({
+    hideDeps: true,
+    limit: 0,
+    summaryOnly: true,
+  }),
+];
+
+const config = [
+  {
```
Comment (wolfgangcodes) [inline rollup.config.mjs:46]: Named variables?

File: examples/hello-world-cjs/index.cjs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const configDefaults = {
+  ignoreNetworkEvents: true,
+};
+
+const main = () => {
+  // Initialize base OTel WebSDK
+  const sdk = new HoneycombWebSDK({
+    // endpoint: 'http://localhost:4318', // send to collector
+    // To send to collector, comment out API Key
+    apiKey: 'hcaik_01hxc8g4cdhrywa0n9132mhfwh6yekhecstp429k90rec31qhv5cfjn5wk',
```
Comment (wolfgangcodes) [inline examples/hello-world-cjs/index.cjs:15]:
```suggestion
    apiKey: 'api-key',
```

**Review by MustafaHaddara — APPROVED** [review]
everything looks good to me! 

do we want the cjs and react examples to make network requests like the hello-world-web one does?

**Comment by wolfgangcodes** [comment]
> do we want the cjs and react examples to make network requests like the hello-world-web one does?
I'm not opposed, but I don't think we _need_ to as part of this PR.

**Review by jairo-mendoza — APPROVED** [review]
Ran all four example apps and they worked great. Creating a new react app from scratch also worked. Looks good to me.

extreme nit: some inconsistencies in READMEs. Example: README for `hello-world-cjs` example says to make changes to the `index.js` file when there's only a `index.cjs` file.

File: examples/hello-world-cjs/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Hello World Example for the Honeycomb Web SDK
+
+You can run this example to see the SDK in action.
+
+## Run this application
+
+`npm install`
+
+Paste your API key into `index.js`, where is says "your api key goes here".
```
Comment (wolfgangcodes) [inline examples/hello-world-cjs/README.md:9]:
```suggestion
Paste your API key into `index.cjs`, where is says "your api key goes here".
```

File: examples/hello-world-cjs/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+`npm install`
+
+Paste your API key into `index.cjs`, where is says "your api key goes here".
+
+`npm run start`
+
+Visit [http://localhost:8080]() to see "hello world".
+
+Now go to [https://ui.honeycomb.io](), click Home, and choose the dataset "web-distro" (unless you changed the `serviceName` in `index.js`).
```
Comment (wolfgangcodes) [inline examples/hello-world-cjs/README.md:15]:
```suggestion
Now go to [https://ui.honeycomb.io](), click Home, and choose the dataset "web-distro" (unless you changed the `serviceName` in `index.cjs`).
```

**Comment by wolfgangcodes** [comment]
> Ran all four example apps and they worked great. Creating a new react app from scratch also worked. Looks good to me.
> 
> extreme nit: some inconsistencies in READMEs. Example: README for `hello-world-cjs` example says to make changes to the `index.js` file when there's only a `index.cjs` file.

Updated in [01cd5bb](https://github.com/honeycombio/honeycomb-opentelemetry-web/pull/175/commits/01cd5bb240f3b6e6477bfa2ef6033fbe0ca07a7f) and [9a07dfc](https://github.com/honeycombio/honeycomb-opentelemetry-web/pull/175/commits/9a07dfcadf18eaeeca3abb231829e34e20c06b62)

**Pull request merged**